pkg/image/api/helper.go

```diff
@@ -157,12 +157,28 @@ func (r DockerImageReference) String() string {
 		registry += "/"
 	}
 
-	if len(r.Namespace) == 0 {
-		r.Namespace = DockerDefaultNamespace
+	namespace := r.Namespace
```

Comment:
This method is fine, rather.  It's the subsequent method that needs to be reverted.